fpdf/template.py

```diff
@@ -60,80 +38,105 @@ def __init__(
             "W": self.write,
         }
         self.texts = {}
-        pdf = self.pdf = FPDF(format=format, orientation=orientation, unit=unit)
-        pdf.set_title(title)
-        pdf.set_author(author)
-        pdf.set_creator(creator)
-        pdf.set_subject(subject)
-        pdf.set_keywords(keywords)
 
     def load_elements(self, elements):
         """Initialize the internal element structures"""
-        self.pg_no = 0
         self.elements = elements
-        self.keys = [v["name"].lower() for v in self.elements]
+        self.keys = []
+        for e in elements:
+            # priority is optional, but we need a default for sorting.
+            if not "priority" in e:
+                e["priority"] = 0
+            for k in ("name", "type", "x1", "y1", "y2"):
+                if k not in e:
+                    raise KeyError(f"Mandatory key '{k}' missing in input data")
+            # x2 is optional for barcode types, but needed for offset rendering
+            if "x2" not in e:
+                if e["type"] in ["B", "C39"]:
+                    e["x2"] = 0
+                else:
+                    raise KeyError("Mandatory key 'x2' missing in input data")
+            self.keys.append(e["name"].lower())
 
     @staticmethod
     def _parse_colorcode(s):
         """Allow hex and oct values for colors"""
-        s = s.strip()
-        if not s:
-            raise ValueError("Foreground and Background must be numeric")
         if s[:2] in ["0x", "0X"]:
             return int(s, 16)
-        if s[0] == "0":
+        if s[:2] in ["0o", "0O"]:
```

Comment:
Why this change?

Reply:
Looks like I got confused by this:
>>> 0123456
SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

I remembered octal literals to just use a zero prefix, not sure when that changed.
And then I just expected int(s, 8) to do the same...
But apparently:
>>> int('0123456', 8)
42798
>>> int('0o123456', 8)
42798

So yeah, I'm going to change that back.